util/conditions/setter_test.go

```diff
@@ -180,6 +184,22 @@ func TestSetLastTransitionTime(t *testing.T) {
 				g.Expect(lastTransitionTime).ToNot(Equal(x))
 			},
 		},
+		{
+			name: "Set a condition that already exists but with different state should preserve the last transition time if defined",
+			to:   setterWithConditions(fooWithLastTransitionTime),
+			new:  fooWithAnotherStateWithLastTransitionTime,
+			LastTransitionTimeCheck: func(g *WithT, lastTransitionTime metav1.Time) {
+				g.Expect(lastTransitionTime).To(Equal(y))
+			},
+		},
+		{
+			name: "Set a condition that already exists but with different Message should preserve the last transition time",
+			to:   setterWithConditions(fooWithLastTransitionTime),
+			new:  fooWithBarMessage,
```

Comment:
What about adding also a test where the new condition has a last transition time time set (but it gets ignored)

Reply:
Added a new test in separate commit, Please let me know if anything else needs to be added. Thank you.